python/cuml/dask/ensemble/randomforestregressor.py

```diff
@@ -318,7 +256,8 @@ def predict(self, X, predict_model="GPU", algo='auto',
         return preds
 
     def predict_using_fil(self, X, delayed, **kwargs):
-        self.local_model = self._concat_treelite_models()
+        if self.local_model is None:
```

Comment:
(Not for this change, just a question for later) Maybe we should rename local_model to "local_concatenated_model" or "local_inference_model" to make it clear that this isn't for fitting but just inference?

Reply:
The self.local_model is the variable that is used by the lazy predict and predict_proba functions. 
The `DelayedParallelFunc` class is used by `DelayedPredictionMixin` and `DelayedPredictionProbaMixin` classes.  `DelayedParallelFunc` reads the `self.local_model` variable to obtain the model which is passed to all the workers for running predict.

Reply:
@JohnZed indeed, the local model is also important for the saving/loading. It would definitely be nice, though, if that distinction were made. Perhaps we could alias `self.local_model` to point to a `self.concatenated_model` to be more concise and readable.

Reply:
Ok, so the naming convention is required here. I think we should keep it as local_model but just find a place to document that in future PRs.